### ISSUE TITLE
Null pointer might occur here (False positive)

### ISSUE THREAD

**Issue by Richard87** [kalessil/phpinspectionsea#1553]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.4.1|
| Language level | 7.4       |

Current behaviour
Doctrine Criteria, `Criteria::expr()` never returns null:
![image](https://user-images.githubusercontent.com/5749715/80689154-2a403080-8acd-11ea-9bb6-5ce7efaae4a7.png)

And the code for `Criteria::expre()`:
```php
    /**
     * Returns the expression builder.
     *
     * @return ExpressionBuilder
     */
    public static function expr()
    {
        if (self::$expressionBuilder === null) {
            self::$expressionBuilder = new ExpressionBuilder();
        }

        return self::$expressionBuilder;
    }
```

Expected behaviour
--
No warning

Environment details
---

![image](https://user-images.githubusercontent.com/5749715/80689261-58257500-8acd-11ea-8245-3d5992b5572a.png)

Might be related to https://github.com/kalessil/phpinspectionsea/issues/1552

Also, thanks for the awesome work!

**Comment by donquixote** [comment]
I can isolate this to a single-file project.

```php
<?php

class C {

  /**
   * Changing this var doc to "self" instead of "self|null" removes the warning.
   *
   * @var self|null
   */
  private static $instance;

  /**
   * @return self
   */
  public static function getInstance() {
    return self::$instance
      ?: self::$instance = new self();
  }

  public function fooStatic() {
    $obj = self::getInstance();
    // Inspection warning below:
    // "[EA] Null pointer exception may occur here."
    $obj->foo();
  }

  public function foo() {

  }

}
```

**Comment by donquixote** [comment]
I'd say the solution should be:
- Within `::getInstance()`, consider only the currently possible type of `self::$instance`, not the annotated type of `self::$instance`.
- Within `::fooStatic()`, consider only the annotated return type of `self::getInstance()`, ignore what is happening within `::getInstance()`. If for some reason `self::getInstance()` returns NULL, that's a bug within that method, not in `::fooStatic()`.

**Comment by donquixote** [comment]
> PhpStorm 2020.1.4
> Build #PS-201.8743.18, built on July 22, 2020
> Runtime version: 11.0.7+10-b765.65 amd64
> VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
> Linux 4.15.0-112-generic
> GC: ParNew, ConcurrentMarkSweep
> Memory: 1904M
> Cores: 8
> Registry: run.processes.with.pty=TRUE, search.everywhere.settings=true, ide.tooltip.initialDelay=952
> Non-Bundled Plugins: R4Intellij, de.thomasrosenau.diffplugin, net.seesharpsoft.intellij.plugins.csv, org.intellij.RegexpTester, Dart, com.dmarcotte.handlebars, com.aurimasniekis.phppsr4namespacedetector, com.kalessil.phpStorm.phpInspectionsEA, com.thanosp.phpstorm.inheritdoc, de.espend.idea.php.annotation, lv.midiana.misc.idea-plugins.deep-js-completion, org.sylfra.idea.plugins.linessorter
> Current Desktop: MATE

**Comment by ea-inspections-team** [comment]
Thank you for reporting, @Richard87! This inspection appears to be somewhat unreliable, so we are disabling it in e1263f8b0607a1f50598cc49c6dbfdcdd73f4baa for now.
